src/iinit.cpp

```diff
@@ -423,131 +414,38 @@ int main( int argc, char **argv )
         return 2;
     }
 
-    // =-=-=-=-=-=-=-
-    // PAM auth gets special consideration, and also includes an
-    // auth by the usual convention
-    bool pam_flg = false;
-    const auto use_legacy_authentication = irods_auth::use_legacy_authentication(*Conn);
-    if (use_legacy_authentication && irods::AUTH_PAM_SCHEME == lower_scheme) {
-        // =-=-=-=-=-=-=-
-        // set a flag stating that we have done pam and the auth
-        // scheme needs overridden
-        pam_flg = true;
-
-        // =-=-=-=-=-=-=-
-        // build a context string which includes the ttl and password
-        std::stringstream ttl_str;  ttl_str << ttl;
-        irods::kvp_map_t ctx_map;
-        ctx_map[irods::AUTH_TTL_KEY] = ttl_str.str();
-        ctx_map[irods::AUTH_PASSWORD_KEY] = "";
-        std::string ctx_str = irods::escaped_kvp_string(ctx_map);
-        // =-=-=-=-=-=-=-
-        // pass the context with the ttl as well as an override which
-        // demands the pam authentication plugin
-        status = clientLogin( Conn, ctx_str.c_str(), irods::AUTH_PAM_SCHEME.c_str() );
-        if ( status != 0 ) {
-            print_error_stack_to_file(Conn->rError, stderr);
-            return 8;
-        }
+    auto ctx = nlohmann::json{
+        {irods::AUTH_TTL_KEY, std::to_string(ttl)},
+        {irods_auth::force_password_prompt, true}
+    };
+
+    // Use the scheme override here to ensure that the authentication scheme in the environment is the same as
+    // the authentication scheme configured here. If the scheme in the environment and the scheme configured in
+    // iinit match, then nothing will need to change in clientLogin. If they do not match, the override wins.
+    if (const int ec = clientLogin(Conn, ctx.dump().c_str(), my_env.rodsAuthScheme); ec != 0) {
+        print_error_stack_to_file(Conn->rError, stderr);
+        rcDisconnect(Conn);
+        return 7;
```

Comment:
are these return codes defined anywhere?

Reply:
Not sure. Did you have something in mind?

Reply:
Yeah, okay.   I don't think we can change them...

So, maybe we just put a comment near each one or a list at the top of where each one is / what it means.   Just to give clarity for the next reader about this question.

Reply:
Added comment just above `main` definition with guide to return values. We can open a new conversation to hash out wording if you'd like. Resolving this one.